Design/Workbench/design-projectsaving.md

```diff
@@ -0,0 +1,115 @@
+﻿
+Project Saving for Mantid Workbench
+==============================
+Motivation
+--------------
+In MantidPlot it was possible to save all currently open windows and all workspaces, and then be able to load then back in the exact(-ish) state that they were in previously. To achieve this goal again in Mantid Workbench, the problem will be split into two where window serialization and saving and therefore reopening will be separate in implementation, from saving workspaces.
+
+The reason for the split is so that it can be handled in a more abstract manor. Instead of thinking of the task as one great big issue it can be split into the two smaller problems and then split further from there, later on in design.
+
+This document exists because Project Saving is a non-trivial challenge to tackle when it comes to ideas with saving and loading interfaces created in both Python and C++ as well as saving and loading of workspaces. This issue should be fairly simple to understand, simply users should be able to save their projects and load them back in, to at least the standard that MantidPlot was able to achieve.
+
+Changes to MantidWorkbench needed
+-----------------------------------------------
+- MainWindow should track all of the open windows (as well as if they are serializable or not) similar to MantidPlot
```

Comment:
I feel a bit like this is making the workbench head back in the direction of `MantidPlot` where a lot of information from application window was required to know what could be serialized. I agree we need access to the various widgets/windows from `mainwindow` but can we leave it up to them to decide what to do when asked to save/load.
The workbench currently uses widgets from the `mantidqt` package through a [plugin mechanism](https://github.com/mantidproject/mantid/blob/master/qt/applications/workbench/workbench/plugins/base.py). This allows the standalone widgets to exist and be resuable by other applications while keeping workbench-specific behaviour inside the workbench.

Reply:
Instead we should probably track this in the plugin mechanism you linked. Thus still allowing us to know what is open and what isn't etc. It doesn't have to be in that central place.